Engine/Helper.cs

```diff
@@ -65,7 +65,10 @@ internal set
             {
                 lock (syncRoot)
                 {
-                    instance = value;
+                    if (instance == null)
```

Comment:
Yes it did cause some issue earlier - `Get-ScriptAnalyzer` command would instantiate `Helper.Instance` to `new Helper()` and when `Invoke-ScriptAnalyzer` would try to set the instance again, it would ignore the setter, which cause `Invoke-ScriptAnalyzer` to crash. 

But after adding [this](https://github.com/PowerShell/PSScriptAnalyzer/blob/c7fa7e1161cef9a38b0c4a3270239ffd6331afed/Engine/Commands/GetScriptAnalyzerRuleCommand.cs#L94), that bug was resolved and the removing this null check was redundant.